appcelerator-modules/ti.map
# fix(ios): polyline click should only fire on line segments and support 2 points

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28527

**Summary:**
- Refactored polyline click handling to only work on line segments like Android.
  * Used to turn polyline into a shape to determine touch points, causing empty areas to wrongly fire clicks.
- Fixed polyline to support click events if it only has 2 points. (Used to require at least 3 points.)

**Test:**
1. Build and run the below on iOS.
2. Tap on the line and verify a dialog appears. _(Verifies 2 point polyline fix.)_
3. Uncomment the 3 lines of code below and re-rerun on iOS.
4. Verify tapping on the different line segments show a dialog.

```javascript
const Map = require('ti.map');
const win = Ti.UI.createWindow();
const polyline1 = Map.createPolyline({
	points: [
		{ latitude: -33.884717, longitude: 151.187993 },  // Sydney
//		{ latitude: -33.882152, longitude: 151.203099 },  // Sydney
		{ latitude: -33.886783, longitude: 151.218033 },  // Sydney
//		{ latitude: 37.7749, longitude: -122.4194 },  // San Francisco
//		{ latitude: 51.5074, longitude: 0.1278 },  // London
	],
	strokeColor: '#FF0000',
	strokeWidth: 4
});
const map = Map.createView({
	region: {
		latitude: -33.87365,
		longitude: 151.20689,
		latitudeDelta: 0.05,
		longitudeDelta: 0.05
	}
});
map.addPolyline(polyline1);
map.addEventListener('click', function(e) {
	let message = 'Click event data:\n';
	message += `- e.clicksource: ${e.clicksource}\n`;
	message += `- e.latitude: ${e.latitude}\n`;
	message += `- e.longitude: ${e.longitude}\n`;
	alert(message);
});
win.add(map);
win.open();
```
